src/wp-includes/i18n/class-wp-i18n-translation-controller.php

```diff
@@ -0,0 +1,404 @@
+	 * Map of loaded translations per locale and text domain.
+	 *
+	 * [ Locale => [ Textdomain => [ .., .. ] ] ]
+	 *
+	 * @since 6.5.0
+	 *
+	 * @var array<string, array<string, WP_I18n_Translation_File[]>>
+	 */
+	protected $loaded_translations = array();
+
+	/**
+	 * List of loaded translation files.
+	 *
+	 * [ Filename => [ Locale => [ Textdomain => WP_I18n_Translation_File ] ] ]
+	 *
+	 * @since 6.5.0
+	 *
+	 * @var array<string,array<string, array<string, WP_I18n_Translation_File|false>>>
+	 */
+	protected $loaded_files = array();
+
+	/**
+	 * Returns the WP_I18n_Translation_Controller singleton.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @return WP_I18n_Translation_Controller
+	 */
+	public static function instance() {
+		static $instance;
+
+		if ( ! $instance ) {
+			$instance = new WP_I18n_Translation_Controller();
+		}
+
+		return $instance;
+	}
+
+	/**
+	 * Returns the current locale.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @return string Locale.
+	 */
+	public function get_locale() {
+		return $this->current_locale;
+	}
+
+	/**
+	 * Sets the current locale.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @param string $locale Locale.
+	 */
+	public function set_locale( string $locale ) {
+		$this->current_locale = $locale;
+	}
+
+	/**
+	 * Loads a translation file.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @param string $translation_file Translation file.
+	 * @param string $textdomain       Text domain.
+	 * @param string $locale           Optional. Locale. Default current locale.
+	 * @return bool True on success, false otherwise.
+	 */
+	public function load( string $translation_file, string $textdomain = 'default', string $locale = null ) {
+		if ( null === $locale ) {
+			$locale = $this->current_locale;
+		}
+
+		$translation_file = realpath( $translation_file );
+
+		if ( false === $translation_file ) {
+			return false;
+		}
+
+		if (
+			isset( $this->loaded_files[ $translation_file ][ $locale ][ $textdomain ] ) &&
+			false !== $this->loaded_files[ $translation_file ][ $locale ][ $textdomain ]
+		) {
+			return false === $this->loaded_files[ $translation_file ][ $locale ][ $textdomain ]->error();
+		}
+
+		if (
+			isset( $this->loaded_files[ $translation_file ][ $locale ] ) &&
+			array() !== $this->loaded_files[ $translation_file ][ $locale ]
+		) {
+			$moe = reset( $this->loaded_files[ $translation_file ][ $locale ] );
+		} else {
+			$moe = WP_I18n_Translation_File::create( $translation_file );
+			if ( false === $moe || false !== $moe->error() ) {
+				$moe = false;
+			}
+		}
+
+		$this->loaded_files[ $translation_file ][ $locale ][ $textdomain ] = $moe;
+
+		if ( ! $moe instanceof WP_I18n_Translation_File ) {
+			return false;
+		}
+
+		if ( ! isset( $this->loaded_translations[ $locale ][ $textdomain ] ) ) {
+			$this->loaded_translations[ $locale ][ $textdomain ] = array();
+		}
+
+		$this->loaded_translations[ $locale ][ $textdomain ][] = $moe;
+
+		return true;
+	}
+
+	/**
+	 * Unload all translation files or a specific one for a given text domain.
```

Comment:
As per docblock standards, let's use third-person singular verb: 
`Unloads all translation files or a specific one for a given text domain.`